auslib/db.py

```diff
@@ -1085,7 +1088,38 @@ def updateRelease(self, name, changed_by, old_data_version, product=None, read_o
             if self.containsForbiddenDomain(blob, product):
                 raise ValueError("Release blob contains forbidden domain.")
             what['data'] = blob.getJSON()
-        self.update(where=[self.name == name], what=what, changed_by=changed_by, old_data_version=old_data_version, transaction=transaction)
+        try:
+            self.update(where=[self.name == name], what=what, changed_by=changed_by, old_data_version=old_data_version, transaction=transaction)
+        except OutdatedDataError as e:
+            if blob is not None and read_only is None:
+                ancestor_blob = createBlob(self.history.getChange(data_version=old_data_version,
+                                                                  column_values={'name': name},
+                                                                  transaction=transaction)
+                                                       .get('data'))
+                # if we have no historical information about the ancestor blob
+                if ancestor_blob is None:
+                    raise
+                tip_release = self.getReleases(name=name,
+                                               transaction=transaction)[0]
+                tip_blob = tip_release.get('data')
+                m = merge.Merger(ancestor_blob, tip_blob, blob, {})
+                try:
+                    m.run()
+                    unified_blob = patch(m.unified_patches, ancestor_blob)
```

Comment:
I'm a bit confused about why Merger and patch are both needed. I suspect it's a quirk of the dictdiffer interface, but since it would be good to have a comment explaining what each of these do.